pa4/checker-pa4.ml

```diff
@@ -1212,121 +1241,101 @@ let main () = begin
+                !parents_method_list
         in
 
         (* Output each method in turn (in order of appearance, with inherited or overridden methods from a superclass coming first; internal methods are defined to appear in ascending alphabetical order) *)
         let output_parent_methods parents_methods_tbl  =
-            let sorted_methods = get_method_names_from_tbl parents_methods_tbl in
-            List.iter (fun method_name ->
-                    let formals, owner_class, exp = (Hashtbl.find parents_methods_tbl method_name) in
-                    fprintf f_out "%s\n" method_name;
-                    fprintf f_out "%d\n" (List.length formals);
-                    output_formal_name formals;
-                    fprintf f_out "%s\n" owner_class;
-                    fprintf f_out "!TODO: expression\n";
-            ) sorted_methods
+                List.iter (fun method_name ->
+                        let method_info_option = (Hashtbl.find_opt parents_methods_tbl method_name) in
+                        if not (Option.none = method_info_option) then begin
+                                let formals, owner_class, exp = Option.get method_info_option in
+                                fprintf f_out "%s\n" method_name;
+                                fprintf f_out "%d\n" (List.length formals);
+                                output_formal_name formals;
+                                fprintf f_out "%s\n" owner_class;
+                                if not (exp.static_type = None) then output_exp exp
+                                else fprintf f_out "expression is None, %s\n" owner_class
+                        end     
+                ) !ordered_method_names
         in
 
         (* print out methods of the current class in the order they appear in source filess *)
-        let output_own_methods current_class own_features parents_method_list = 
-            List.iter (fun current_method ->
-                    match current_method with
-                    | Attribute _ -> ()
-                    | Method ((_,method_name), formals, method_type, exp) -> begin
-                            fprintf f_out "%s\n" method_name;
-                            fprintf f_out "%d\n" (List.length formals);
-                            output_formal_name formals;
-                            if not (List.mem method_name parents_method_list) then begin
-                                    fprintf f_out "%s\n" current_class;
-                                    if not (exp.static_type = None) then output_exp exp
-                                    else fprintf f_out "expression is None\n"
-                            end;
-                    end
-            ) own_features
+        let output_own_methods current_class own_features parents_methods_tbl = 
+                List.iter (fun current_method ->
+                        match current_method with
+                        | Attribute _ -> ()
```

Comment:
We could put this under Method and make it `| _ -> ()` in order to make this pattern match exhaustive while still having the same functionality.